humanmine.ipynb

```diff
@@ -331,206 +341,390 @@
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
     }
    ],
    "source": [
-    "print(projects)"
+    "html = \"<table>\"\n",
+    "for gene in development_stage:\n",
+    "    images = development_stage[gene]\n",
+    "    if len(images) > 0:\n",
+    "        html += '<tr><td><h2>Gene: '+gene+'</h2></td></tr><tr>'\n",
+    "        for k, v in images:\n",
+    "            html += '<tr><td><h4>Developmental stage: '+k+'</h4></td></tr><tr>'\n",
+    "            for i in v:\n",
+    "                qs = {'id': i}\n",
+    "                url = IMAGE_DATA_URL.format(**qs)\n",
+    "                url_link = LINK_URL.format(**qs)\n",
+    "                html += '<td><a href=\"'+url_link+'\" target=\"_blank\"><img src=\"'+url+'\"/></a></td>'\n",
+    "            html += \"</tr>\"\n",
+    "        html += \"</tr>\"\n",
+    "html += \"</table>\"\n",
+    "display(HTML(html))"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "fallen-girlfriend",
+   "id": "367f36ee",
    "metadata": {},
    "source": [
-    "## Find the images\n",
-    "Find the images linked to the selected genes in one the project found.\n",
-    "Below, we look at the gene **PDX1**"
+    "**Now we filter images with a given Organism Part value**.\n",
+    "\n",
+    "We explore the images associated to the gene **PDX1** with abnormal pathology status."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 17,
-   "id": "little-monkey",
+   "execution_count": 165,
+   "id": "55f1cf9b",
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "defaultdict(<class 'list'>, {'15PCW': [9841210, 9841211], '9PCW': [9841212, 9841219, 9841213, 9841218], 'CS16': [9841216, 9841217], 'CS21': [9841215, 9841214]})\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
-    "from collections import defaultdict\n",
-    "project_id = 1104\n",
-    "selected_genes = {\"PDX1\"}\n",
-    "images = defaultdict(list)\n",
-    "for gene in selected_genes:\n",
-    "    qs3 = {'key': TYPE, 'value': gene, 'project_id': project_id}\n",
-    "    url3 = DATASETS_URL.format(**qs3)\n",
-    "    find_images(session.get(url3).json(), \"datasets\")\n",
-    "\n",
-    "print(images)"
+    "pathology_images = {}\n",
+    "for k in results:\n",
+    "    images = results[k]\n",
+    "    result_images = defaultdict(list)\n",
+    "    for image in images:\n",
+    "        values = image[\"key_values\"]\n",
+    "        part = None\n",
+    "        for v in values:\n",
+    "            name = v[\"name\"]\n",
+    "            value = v['value']\n",
+    "            if PART in name and (TISSUE or EXPRESSION in value):\n",
```

Comment:
This line throws an error

```
NameError                                 Traceback (most recent call last)
/tmp/ipykernel_61/654637249.py in <module>
      9             name = v["name"]
     10             value = v['value']
---> 11             if PART in name and (TISSUE or EXPRESSION in value):
     12                 part = value
     13         for v in values:

NameError: name 'PART' is not defined
```

Reply:
(I tried to fix it by de-capitalizing the undefined PART variable, but this then threw error complaining about the ``part`` being None.

Reply:
Aha, sorry, got it, it should be ``PART_KEY`` as defined in the top of the notebook.
Will make a suggestion.